controllers/replica_set_controller.go

```diff
@@ -541,6 +541,7 @@ func buildAutomationConfig(mdb mdbv1.MongoDBCommunity, auth automationconfig.Aut
 	}
 
 	return automationconfig.NewBuilder().
+		IsEnterprise(os.Getenv(construct.MongodbImageEnv) == construct.OfficialMongodbEnterpriseServerImageName).
```

Comment:
This actually might not be true, because our environment variable (at least in enterprise)  does not have to be the source of truth, since we can define a wrapper in our crd. 💀 

For internal reference: https://github.com/10gen/ops-manager-kubernetes/blob/25eacd7470ef05bf699e4eea185757c482a36614/api/v1/om/appdb_types.go#L258-L275 + https://github.com/10gen/ops-manager-kubernetes/blob/a68a1f281e297eec893ea936c9f2316722aefc8b/docker/mongodb-enterprise-tests/tests/opsmanager/withMonitoredAppDB/om_appdb_configure_all_images.py#L36-L51

Reply:
I implemented a piece of logic with `guessEnterprise`. It checks if there's an override via env. variable and if it's missing - it tries to guess the value. I think with this approach we should be on the safe side as you can always override to bring it to desired behavior.